Source/Core/DolphinQt2/CheatsManager.cpp

```diff
@@ -482,139 +476,134 @@ void CheatsManager::Update()
     return;
   }
 
-  m_updating = true;
-
   m_result_label->setText(tr("%1 Match(es)").arg(m_results.size()));
   m_match_table->setRowCount(static_cast<int>(m_results.size()));
 
-  for (size_t i = 0; i < m_results.size(); i++)
-  {
-    auto* address_item = new QTableWidgetItem(
-        QStringLiteral("%1").arg(m_results[i].address, 8, 16, QLatin1Char('0')));
-    auto* value_item = new QTableWidgetItem;
+  if (m_results.empty())
+    return;
 
-    address_item->setFlags(Qt::ItemIsEnabled | Qt::ItemIsSelectable);
-    value_item->setFlags(Qt::ItemIsEnabled | Qt::ItemIsSelectable);
+  m_updating = true;
 
-    if (PowerPC::HostIsRAMAddress(m_results[i].address))
+  Core::RunAsCPUThread([this] {
+    for (size_t i = 0; i < m_results.size(); i++)
     {
-      const auto prev_state = Core::GetState();
-      Core::SetState(Core::State::Paused);
+      auto* address_item = new QTableWidgetItem(
+          QStringLiteral("%1").arg(m_results[i].address, 8, 16, QLatin1Char('0')));
+      auto* value_item = new QTableWidgetItem;
+
+      address_item->setFlags(Qt::ItemIsEnabled | Qt::ItemIsSelectable);
+      value_item->setFlags(Qt::ItemIsEnabled | Qt::ItemIsSelectable);
 
-      switch (m_results[i].type)
+      if (PowerPC::HostIsRAMAddress(m_results[i].address))
       {
-      case DataType::Byte:
-        value_item->setText(QStringLiteral("%1").arg(PowerPC::HostRead_U8(m_results[i].address), 2,
-                                                     16, QLatin1Char('0')));
-        break;
-      case DataType::Short:
-        value_item->setText(QStringLiteral("%1").arg(PowerPC::HostRead_U16(m_results[i].address), 4,
-                                                     16, QLatin1Char('0')));
-        break;
-      case DataType::Int:
-        value_item->setText(QStringLiteral("%1").arg(PowerPC::HostRead_U32(m_results[i].address), 8,
-                                                     16, QLatin1Char('0')));
-        break;
-      case DataType::Float:
-        value_item->setText(QString::number(PowerPC::HostRead_F32(m_results[i].address)));
-        break;
-      case DataType::Double:
-        value_item->setText(QString::number(PowerPC::HostRead_F64(m_results[i].address)));
-        break;
-      case DataType::String:
-        value_item->setText(tr("String Match"));
-        break;
+        switch (m_results[i].type)
+        {
+        case DataType::Byte:
+          value_item->setText(QStringLiteral("%1").arg(PowerPC::HostRead_U8(m_results[i].address),
+                                                       2, 16, QLatin1Char('0')));
+          break;
+        case DataType::Short:
+          value_item->setText(QStringLiteral("%1").arg(PowerPC::HostRead_U16(m_results[i].address),
+                                                       4, 16, QLatin1Char('0')));
+          break;
+        case DataType::Int:
+          value_item->setText(QStringLiteral("%1").arg(PowerPC::HostRead_U32(m_results[i].address),
```

Comment:
Are you allowed to do this from the CPU thread?

Reply:
No, it frequently causes issues.